### PR TITLE
update handlebars engine

### PR DESCRIPTION
**Pull request by gmarinov**
- Handlebars 4
- include helpers & partials in templateCache
- add diy-hbs-helpers to Handlebars.helpers
- reference Handlebars.partials correctly
- move hbs-helpers.js from utils/ to lib/

screen shot 2016-06-27 at 17 17 24

p.s. remember to rebuild your installation or update handlebars, gulp-handlebars and express-handlebars

### CODE REVIEW

File: views/server/partials/gtm.handlebars [hunk 1]
```diff
@@ -20,5 +20,5 @@ height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
 new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
 j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
 '//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
-})(window,document,'script','dataLayer','{{{KEY_GTM}}}');</script>
+})(window,document,'script','dataLayer','{{{KEY_GTM}}}');</scr{{undefined}}ipt>
```
Comment (kschzt) [inline views/server/partials/gtm.handlebars]: oops?

File: views/server/partials/home/aboutIntro.handlebars [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <section id="aboutIntro">
     <div class="in">
         <section id="aboutEmbed">
-            <script src="/scripts/embed.js" data-aspect="0.95" data-vizorurl="/embed/matt/hmd-bake-spinner?autoplay=true&noheader=true"></script>
+            <script src="/scripts/embed.js" data-aspect="0.95" data-vizorurl="/embed/matt/hmd-bake-spinner?autoplay=true&noheader=true"></scr{{undefined}}ipt>
```
Comment (kschzt) [inline views/server/partials/home/aboutIntro.handlebars]: oops?

File: views/server/partials/gtm.handlebars [hunk 3]
```diff
@@ -11,7 +11,7 @@
 		dataLayer.push(evt)
 	}
 
-</script>
+</scr{{undefined}}ipt>
```
Comment (kschzt) [inline views/server/partials/gtm.handlebars]: oops?

Comment (gmarinov) [inline views/server/partials/gtm.handlebars]: this is a known handlebars issue, and a common way to resolve it.

File: lib/templateCache.js [hunk 4]
```diff
@@ -3,21 +3,37 @@ var handlebars = require('gulp-handlebars')
...
 
 var cache = ''
+var partials=''
+var helpers=''
+var opts = {}
 
 function TemplateCache() {}
 
 TemplateCache.prototype.recompile = function(cb) {
 	cache = ''
```
Comment (kschzt) [inline lib/templateCache.js]: can you rename cache to something else then

File: app.js [hunk 5]
```diff
@@ -70,18 +70,23 @@ app.events = new EventEmitter()
...
 		{dir:'views/server/partials', namespace: 'srv'}
 	],
 	helpers: _.extend(
-		hbsHelpers,
 		diyHbsHelpers,
+		hbsHelpers,
 		templateCache.helper()
 	)
 })
+templateCache.setHbs(hbs.handlebars)	// work around derp gulp-handlebars
```
Comment (kschzt) [inline app.js]: pls remove comment

File: lib/templateCache.js [hunk 6]
```diff
@@ -3,21 +3,37 @@ var handlebars = require('gulp-handlebars')
...
+	partials = ''
+	helpers = ''
 	return this.compile(cb)
 }
+TemplateCache.prototype.setHbs = function(hbs) {
+	opts.handlebars = hbs
```
Comment (kschzt) [inline lib/templateCache.js]: `opts` is not local to this class, setting this changes it for all instances of TemplateCache. How about using `this.hbsOptions = { handlebars: hbs }`?

**Comment by kschzt** [comment]
lgtm

**Pull request merged**